pkg/test/e2e/cilium/cilium_test.go

```diff
@@ -0,0 +1,660 @@
+				}
+
+				allHealthy = allHealthy && podHealthy
+			}
+
+			if !allHealthy {
+				t.Logf("not all pods healthy yet...")
+				return false, nil
+			}
+
+			t.Logf("all hubble pods healthy")
+
+			return true, nil
+		})
+		if err != nil {
+			t.Fatalf("pods never became ready: %v", err)
+		}
+	}
+
+	t.Logf("test hubble relay observe")
+	{
+		conn, err := grpc.Dial(fmt.Sprintf("%s:%d", nodeIP, 30077), grpc.WithInsecure())
+		if err != nil {
+			t.Fatalf("failed to dial to hubble relay: %v", err)
+		}
+		defer conn.Close()
+
+		nFlows := 20
+		flowsClient, err := observer.NewObserverClient(conn).
+			GetFlows(context.Background(), &observer.GetFlowsRequest{Number: uint64(nFlows)})
+		if err != nil {
+			t.Fatalf("failed to get flow client:%v", err)
+		}
+
+		for c := 0; c < nFlows; c++ {
+			flow, err := flowsClient.Recv()
+			if err != nil {
+				t.Fatalf("failed to get flow:%v", err)
+			}
+			fmt.Println(flow)
+		}
+	}
+
+	t.Logf("test hubble ui observe")
+	{
+		req, err := http.NewRequestWithContext(context.Background(), http.MethodGet,
+			fmt.Sprintf("http://%s:%d", nodeIP, 30007), nil)
+		if err != nil {
+			t.Fatalf("failed to construct request to hubble ui: %v", err)
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatalf("failed to get response from hubble ui: %v", err)
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected: 200 OK, got: %d", resp.StatusCode)
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read response body:%v", err)
+		}
+
+		if !strings.Contains(string(body), "Hubble") {
+			t.Fatalf("failed to find Hubble in the body")
+		}
+	}
+}
+
+func filterByLabel(pods []corev1.Pod, key string, values ...string) []corev1.Pod {
+	c := 0
+	for _, p := range pods {
+		for _, v := range values {
+			vv, ok := p.Labels[key]
+			if ok && vv == v {
+				c++
+				pods = append(pods, p)
+			}
+		}
+	}
+	return pods[len(pods)-c:]
+}
+
+func makeScheme() *runtime.Scheme {
+	var s = runtime.NewScheme()
+	_ = serializer.NewCodecFactory(s)
+	_ = runtime.NewParameterCodec(s)
+	utilruntime.Must(appsv1.AddToScheme(s))
+	utilruntime.Must(corev1.AddToScheme(s))
+	utilruntime.Must(ciliumv2.AddToScheme(s))
+	return s
+}
+
+func resourcesFromYaml(filename string, s *runtime.Scheme) ([]runtime.Object, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	manifests, err := yamlutil.ParseMultipleDocuments(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	sr := kjson.NewSerializerWithOptions(&kjson.SimpleMetaFactory{}, s, s, kjson.SerializerOptions{})
+
+	var objs []runtime.Object
+	for _, m := range manifests {
+		obj, err := runtime.Decode(sr, m.Raw)
+		if err != nil {
+			return nil, err
+		}
+		objs = append(objs, obj)
+	}
+
+	return objs, nil
+}
+
+// creates a usercluster on aws.
+func createUsercluster(t *testing.T, proxyMode string) (string, string, func(), error) {
```

Comment:
I have some questions about this:
1. Second return value is not used, so it can be removed?
2. Why do you need to save the kubeconfig to a file?

Reply:
1. This function was copied from other test TestKonnectivity. It is used there. I will move this function to the util package for use in both the tests. 
2. Hmm weird. Makes not much sense. I did it at the time and don't remember why.

Reply:
Can I do 1. in another PR? Konnectivity tests are disabled. I need to debug them and I am planning to refactor this in that PR. 

Fixed 2.

Reply:
sure, you can do it when the konnectivity tests are being fixed